src/uproot/streamers.py

```diff
@@ -1064,7 +1064,7 @@ def class_code(
                     f'            forth_stash.add_to_header(f"output node{{key}}-data {{uproot._awkward_forth.convert_dtype(uproot._awkward_forth.symbol_dict[self._dtype{len(dtypes)}])}}\\n")',
                     '            forth_stash.add_to_header(f"output node{key2}-offsets int64\\n")',
                     '            forth_stash.add_to_init(f"0 node{key2}-offsets <- stack\\n")',
-                    f'            content[{self.name!r}] = {{"class": "ListOffsetArray", "offsets": "i64", "content": {{ "class": "NumpyArray", "primitive": f"{{uproot._awkward_forth.convert_dtype(uproot._awkward_forth.symbol_dict[self._dtype{len(dtypes)}])}}", "inner_shape": [], "parameters": {{}}, "form_key": f"node{{key}}"}}, "form_key": f"node{{key2}}"}}',
+                    f'            content[{self.name!r}] = {{"class": "RegularArray", "size": {self.array_length}, "content": {{ "class": "NumpyArray", "primitive": f"{{uproot._awkward_forth.convert_dtype(uproot._awkward_forth.symbol_dict[self._dtype{len(dtypes)}])}}", "inner_shape": [], "parameters": {{}}, "form_key": f"node{{key}}"}}, "form_key": f"node{{key2}}"}}',
                     f'            forth_stash.add_to_pre(f"{self.array_length} dup node{{key2}}-offsets +<- stack \\n stream #!{{uproot._awkward_forth.symbol_dict[self._dtype{len(dtypes)}]}}-> node{{key}}-data\\n")\n',
```

Comment:
@jpivarski what I did here was hard-code the array length into the forth code and use it to read the file by putting and pulling it off the stack. I think this can be easily replaced with a similar hardcoded version without generating the offset array.

Reply:
By "hard coded," do you mean that you took the value of `self.array_length` and included that as a literal in the Forth code? (That would be fine, of course.) You don't mean that you took the length of this particular example, such that it wouldn't work with a file with a different array length, right? (Just checking, because of the way it was worded.)

We can remove the offsets-generation at any time, because that counts as an optimization—the output won't change when it gets fixed.

Reply:
The first one. It changes depending on the file.

Reply:
Okay, great! (I thought so, just checking.)

And that reminds me that I should merge this now. Thanks for looking out over!